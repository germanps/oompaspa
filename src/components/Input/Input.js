import React from 'react'
import './Input.scss'
import lens from '../../assets/ic_search.png'
import PropTypes from 'prop-types'

export const Input = ({ handleInputChange }) => {
    const inputLens = {
        backgroundImage: `url(${lens})`,
        backgroundRepeat: "no-repeat",
        backgroundSize: "10px",
        backgroundPosition: "right 5px bottom 5px"
    }
    return (
        <div className="input-wrapper">
            <label htmlFor="search" className="sr-only">BÃºsqueda</label>
            <input
                id="search"
                type="text"
                placeholder="Search"
                name="search"
                onChange={handleInputChange}
                style={inputLens}
                className="control"
                autoComplete="off"
            />
        </div>
    )
}

Input.propTypes = {
    handleInputChange: PropTypes.func.isRequired
}