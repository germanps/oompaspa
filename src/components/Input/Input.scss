@import "../../scss/index.scss";

.input-wrapper{

    .control{
        border: 1px solid $border-grey-dark;
        border-radius: 3px;
        max-width: 150px;
        padding: 0 5px;
        
        &:focus{
            outline: none;
            border: 1px solid $bronze; 
        }
    }
}