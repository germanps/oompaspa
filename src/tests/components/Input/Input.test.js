import React from 'react'
import { shallow } from 'enzyme'
import { Input } from '../../../components/Input/Input'


describe('Test in <Input />', () => {

    const handleInputChange = () => { }
    const wrapper = shallow(
        <Input handleInputChange={handleInputChange} />
    )

    test('should show component correctly with its props ', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
