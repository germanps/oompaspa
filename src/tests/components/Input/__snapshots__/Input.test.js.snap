// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Input /> should show component correctly with its props  1`] = `
<div
  className="input-wrapper"
>
  <label
    className="sr-only"
    htmlFor="search"
  >
    BÃºsqueda
  </label>
  <input
    autoComplete="off"
    className="control"
    id="search"
    name="search"
    onChange={[Function]}
    placeholder="Search"
    style={
      Object {
        "backgroundImage": "url(ic_search.png)",
        "backgroundPosition": "right 5px bottom 5px",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "10px",
      }
    }
    type="text"
  />
</div>
`;
